src/test/kotlin/org/rust/ide/intentions/RemoveDbgIntentionTest.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+
+class RemoveDbgIntentionTest : RsIntentionTestBase(RemoveDbgIntention()) {
```

Comment:
Please provide more tests. For example, with something like `foo(x, dbg!((5 + 7) * 2), y)`

Reply:
👌

Reply:
I have no idea why `caret` will be in this position after remove `dbg!`.

```kotlin
    fun `test remove outer dbg!`() = doAvailableTest("""
        fn test() {
            let a = dbg!(1 +/*caret*/ dbg!(3));
        }
    """, """
        fn test() {
            let a = 1 + dbg!/*caret*/(3);
        }
    """)
```

Reply:
You should move caret after replace, and don't forget to invoke `doPostponedOperationsAndUnblockDocument`:
```Kotlin
override fun invoke(project: Project, editor: Editor, ctx: RsMacroExpr) {
    val expr = ctx.macroCall.exprMacroArgument?.expr ?: return
    val newExpr = ctx.replace(expr)
    PsiDocumentManager.getInstance(project).doPostponedOperationsAndUnblockDocument(editor.document)
    editor.caretModel.moveToOffset(newExpr.textOffset)
}
```

Reply:
Also, please squash your commits into one before merge